File: service-request-backend/src/main/java/com/servicedesk/service/ServiceCatalogService.java
package com.servicedesk.service;

import com.servicedesk.entity.ServiceCatalog;
import com.servicedesk.entity.ServiceCategory;
import com.servicedesk.exception.ResourceNotFoundException;
import com.servicedesk.repository.ServiceCatalogRepository;
import com.servicedesk.repository.ServiceCategoryRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

/**
 * Service Catalog Service
 * Business logic for service catalog management
 */
@Service
@Transactional
public class ServiceCatalogService {

    @Autowired
    private ServiceCatalogRepository serviceCatalogRepository;

    @Autowired
    private ServiceCategoryRepository serviceCategoryRepository;

    public List<ServiceCategory> getAllCategories() {
        return serviceCategoryRepository.findAll();
    }

    public List<ServiceCategory> getActiveCategories() {
        return serviceCategoryRepository.findByIsActive(true);
    }

    public ServiceCategory getCategoryById(Long id) {
        return serviceCategoryRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("ServiceCategory", "id", id));
    }

    public ServiceCategory createCategory(ServiceCategory category) {
        return serviceCategoryRepository.save(category);
    }

    public ServiceCategory updateCategory(Long id, ServiceCategory categoryDetails) {
        ServiceCategory category = getCategoryById(id);
        category.setName(categoryDetails.getName());
        category.setDescription(categoryDetails.getDescription());
        category.setIcon(categoryDetails.getIcon());
        category.setIsActive(categoryDetails.getIsActive());
        return serviceCategoryRepository.save(category);
    }

    public void deleteCategory(Long id) {
        ServiceCategory category = getCategoryById(id);
        serviceCategoryRepository.delete(category);
    }

    public List<ServiceCatalog> getAllServices() {
        return serviceCatalogRepository.findAll();
    }

    public List<ServiceCatalog> getActiveServices() {
        return serviceCatalogRepository.findByIsActive(true);
    }

    public List<ServiceCatalog> getServicesByCategory(Long categoryId) {
        return serviceCatalogRepository.findByCategoryIdAndIsActive(categoryId, true);
    }

    public ServiceCatalog getServiceById(Long id) {
        return serviceCatalogRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("ServiceCatalog", "id", id));
    }

    public ServiceCatalog createService(ServiceCatalog service) {
        return serviceCatalogRepository.save(service);
    }

    public ServiceCatalog updateService(Long id, ServiceCatalog serviceDetails) {
        ServiceCatalog service = getServiceById(id);
        service.setName(serviceDetails.getName());
        service.setDescription(serviceDetails.getDescription());
        service.setCategory(serviceDetails.getCategory());
        service.setSla(serviceDetails.getSla());
        service.setIsActive(serviceDetails.getIsActive());
        service.setRequiresApproval(serviceDetails.getRequiresApproval());
        return serviceCatalogRepository.save(service);
    }

    public ServiceCatalog toggleServiceStatus(Long id) {
        ServiceCatalog service = getServiceById(id);
        service.setIsActive(!service.getIsActive());
        return serviceCatalogRepository.save(service);
    }

    public void deleteService(Long id) {
        ServiceCatalog service = getServiceById(id);
        serviceCatalogRepository.delete(service);
    }
}

File: service-request-frontend/src/services/serviceCatalogService.js
import api from '../api/axios';

/**
 * Service Catalog Service
 * Handles service catalog API calls
 */
const serviceCatalogService = {
    // Admin APIs
    /**
     * Get all services (admin)
     */
    getAllServices: async () => {
        const response = await api.get('/admin/service-catalog');
        return response.data;
    },

    /**
     * Get service by ID
     */
    getServiceById: async (id) => {
        const response = await api.get(`/admin/service-catalog/${id}`);
        return response.data;
    },

    /**
     * Create new service
     */
    createService: async (serviceData) => {
        const response = await api.post('/admin/service-catalog', serviceData);
        return response.data;
    },

    /**
     * Update service
     */
    updateService: async (id, serviceData) => {
        const response = await api.put(`/admin/service-catalog/${id}`, serviceData);
        return response.data;
    },

    /**
     * Delete service (soft delete)
     */
    deleteService: async (id) => {
        const response = await api.delete(`/admin/service-catalog/${id}`);
        return response.data;
    },

    /**
     * Get all categories (admin)
     */
    getAllCategories: async () => {
        const response = await api.get('/admin/service-catalog/categories');
        return response.data;
    },

    /**
     * Create category
     */
    createCategory: async (categoryData) => {
        const response = await api.post('/admin/service-catalog/categories', categoryData);
        return response.data;
    },

    /**
     * Update category
     */
    updateCategory: async (id, categoryData) => {
        const response = await api.put(`/admin/service-catalog/categories/${id}`, categoryData);
        return response.data;
    },

    // User APIs
    /**
     * Get active categories (user)
     */
    getActiveCategories: async () => {
        const response = await api.get('/user/service-catalog/categories');
        return response.data;
    },

    /**
     * Get services by category (user)
     */
    getServicesByCategory: async (categoryId) => {
        const response = await api.get(`/user/service-catalog/categories/${categoryId}/services`);
        return response.data;
    },

    /**
     * Get all active services (user)
     */
    getActiveServices: async () => {
        const response = await api.get('/user/service-catalog/services');
        return response.data;
    }
};

export default serviceCatalogService;
